Pipelines/templates/dotnet-publish-nix-job.yml

```diff
@@ -0,0 +1,96 @@
+parameters:
+# Job Name
+- name: jobName
+  type: string
+  default: 'publish_nix'
```

Comment:
Either this should be named approrpiately (nix-mac-job) or it should only do linux with a separate job that does mac

Reply:
Using nix as a loose term here for Unix/Linux-style systems. We could create a separate job for Mac, but this is a bit faster. Easy enough to split in the future if we see a need for it.

Reply:
Not opposed to renaming the default jobName though.

Reply:
I think just rename the default job. The windows one is called "windows_netcore" or whatever

Reply:
Did this, as well as changed the file name for consistency.